kedro/framework/context/context.py

```diff
@@ -158,18 +158,38 @@ def _expand_full_path(project_path: str | Path) -> Path:
     return Path(project_path).expanduser().resolve()
 
 
-@frozen
+@define(slots=False)  # Enable setting new attributes to `KedroContext`
```

Comment:
Is this still needed now we're not freezing anything?

Reply:
> I was just wondering if you then still need all the tests for checking attributes?

It is needed, and for this reason I keep the test. If you try to set `slots=True` (or remove this argument since it's default), you will see the test fail.  [The default for standard Python library is `False`](https://docs.python.org/3/library/dataclasses.html#dataclasses.dataclass)

https://www.attrs.org/en/stable/api.html#attrs.define - The reason for this is slotted class is more efficient and make sense if it is used as "dataclass" (think of pydantic model that get convert to JSON etc), in this case it's irrelevant